contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,590 @@
+        motion.overallApprovalTimestamp = block.timestamp;
+      }
+    }
+  }
+
+  function cancelFunctionality(uint256 _motionId) internal {
+    motions[_motionId].rejected = true;
+
+    emit MotionCancelled(msgSender(), _motionId);
+  }
+
+  function executeFunctionality(
+    uint256 _motionId,
+    bool _failingAllowedByUser
+  ) internal motionExists(_motionId) notExecuted(_motionId) notRejected(_motionId) {
+    Motion storage motion = motions[_motionId];
+
+    require(checkThreshold(_motionId, Vote.Approve), "multisig-not-enough-approvals");
+
+    // If approvals were made, threshold lowered, and then executed,
+    //  motion.overallApprovalTimestamp is 0 (since it was never set)
+    if (motion.overallApprovalTimestamp == 0) {
+      // We set the overall approval timestamp to now, and return
+      // We don't execute the motion, but we want to commit the timestamp
+      //  (which wouldn't happen if we continued and a call failed)
+      motion.overallApprovalTimestamp = block.timestamp;
+      return;
+    }
+
+    motion.executed = true;
+    bool overallSuccess = true;
+
+    for (uint256 i = 0; i < motion.data.length; i += 1) {
+      // solhint-disable-next-line avoid-low-level-calls
+      (bool success, ) = address(getTarget(motion.targets[i], address(colony))).call(
+        motion.data[i]
+      );
+      overallSuccess = overallSuccess && success;
+
+      // Allow failing execution after seven days, if the user allowed it
+      require(
+        success ||
+          (_failingAllowedByUser && motion.overallApprovalTimestamp + 7 days <= block.timestamp),
+        "multisig-failed-not-one-week"
+      );
+    }
+
+    emit MotionExecuted(msgSender(), _motionId, overallSuccess);
+  }
+
+  function checkThreshold(uint256 _motionId, Vote _vote) internal view returns (bool thresholdMet) {
+    Motion storage motion = motions[_motionId];
+    uint8 roleIndex;
+
+    // While there are still relevant roles we've not checked yet
+    while (uint256(motion.requiredPermissions) >= (1 << roleIndex)) {
+      // For the current role, is it required for the motion?
+      if ((motion.requiredPermissions & bytes32(1 << roleIndex)) > 0) {
+        uint256 threshold = getDomainSkillRoleThreshold(motion.domainSkillId, roleIndex);
+        if (motionVoteCount[_motionId][_vote][roleIndex] < threshold) {
+          return false;
+        }
+      }
+      roleIndex += 1;
+    }
+
+    return true;
+  }
+
+  function validateMotionDomain(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _motionId
+  ) internal view {
+    uint256 permissionSkillId = colony.getDomain(_permissionDomainId).skillId;
+    uint256 domainSkillId = colonyNetwork.getChildSkillId(permissionSkillId, _childSkillIndex);
+
+    // slither-disable-next-line incorrect-equality
+    require(domainSkillId == motions[_motionId].domainSkillId, "multisig-not-same-domain");
+  }
+
+  bytes32 constant ROOT_AND_ARCHITECTURE =
```

Comment:
This should probably be up top next to the other permission masks.

Reply:
It's not actually used any more, so I've deleted it.